src/main/java/tasks/Tasks.java

```diff
@@ -0,0 +1,156 @@
+package tasks;
+
+import java.util.ArrayList;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.File;
+
+public abstract class Tasks {
+    final public static String saveFilesDir = System.getProperty("user.dir") + File.separator + "saveFiles.txt";
+    boolean isCompleted;
+    String name;
+
+    public abstract String getName();
+    public abstract String getTaskData();
+
+    protected void makeComplete() {
+        isCompleted = true;
+    }
+
+    public String getComplete(){
+        if (isCompleted) {
+            return "X";
+        } else {
+            return " ";
+        }
+    }
+
+    //Function to execute function to manage tasks
+    public static void manageTasks(String input, ArrayList<Tasks> tasksAL) {
+        if (input.equalsIgnoreCase("list")) {
+            if (tasksAL.size() == 0){
+                System.out.println("There is nothing in your list! Try adding something into your list.");
+            } else {
+                printTasks(tasksAL);
+            }
+
+        } else if(input.toLowerCase().startsWith("done") && input.split(" ").length == 2) {
+            try {
+                int taskIndex = Integer.parseInt(input.split(" ")[1]) - 1;
+                if (tasksAL.get(taskIndex - 1).getComplete().equals("X")){
+                    System.out.println("You have already completed that task!");
+                } else {
+                    tasksAL.get(taskIndex).makeComplete();
+                    printCompleted(tasksAL, taskIndex);
+                }
+            } catch(IndexOutOfBoundsException e){
+                System.out.println("Number does not exist in list, please try again.");
+            } catch(NumberFormatException e){
+                System.out.println("Please add 'todo', 'deadline' or 'event' in the front of your task!");
+            }
+
```

Comment:
perhaps apply SLAP here? one method for each else if you have(e.g done, delete, todo and etc)